tests/test_group.rs

```diff
@@ -239,36 +241,37 @@ fn group_operations() {
         let bob_key_package_bundle = KeyPackageBundle::new(
             &[ciphersuite_name],
             &bob_credential_bundle,
-            mandatory_extensions,
+            mandatory_extensions.clone(),
         );
         let bob_key_package = bob_key_package_bundle.get_key_package();
 
-        // Alice creates a group
+        // === Alice creates a group ===
         let group_id = [1, 2, 3, 4];
-        let mut group_alice_1234 = MlsGroup::new(
+        let group_config = GroupConfig::default();
+        let mut group_alice = MlsGroup::new(
             &group_id,
             ciphersuite_name,
             alice_key_package_bundle,
-            GroupConfig::default(),
+            group_config,
         );
 
-        // Alice adds Bob
-        let bob_add_proposal = group_alice_1234.create_add_proposal(
+        // === Alice adds Bob ===
+        let bob_add_proposal = group_alice.create_add_proposal(
             group_aad,
             &alice_credential_bundle,
             bob_key_package.clone(),
         );
         let epoch_proposals = vec![bob_add_proposal];
-        let (mls_plaintext_commit, welcome_bundle_alice_bob_option, kpb_option) =
-            match group_alice_1234.create_commit(
+        let (mls_plaintext_commit, welcome_bundle_alice_bob_option, kpb_option) = match group_alice
+            .create_commit(
                 group_aad,
                 &alice_credential_bundle,
                 epoch_proposals.clone(),
```

Comment:
So this didn't change but I'm not quite sure why we have to clone the proposals here (not sure if I asked this before...). If the proposals don't change and we need them again outside, shouldn't they be passed in as reference?

Reply:
The reason why the proposals are cloned here is because we simulate different groups. In a real scenario they would be "cloned" by the Delivery Service.